hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/storefiletracker/StoreFileListFilePrettyPrinter.java

```diff
@@ -0,0 +1,192 @@
+    files.addOption(new Option(regionOption, "region", true, "Region to scan. Pass region name; e.g. '3d58e9067bf23e378e68c071f3dd39eb'"));
+    files.addOption(new Option(fileOption, "file", true, "File to scan. Pass full-path; e.g. hdfs://a:9000/hbase/hbase:meta/12/34"));
+    options.addOptionGroup(files);
+  }
+
+  public boolean parseOptions(String args[]) throws ParseException, IOException {
+    HelpFormatter formatter = new HelpFormatter();
+    if (args.length == 0) {
+      formatter.printHelp("sft [--file=</path/to/tracker/file> | --table=<namespace:tablename|tablename> --region=<regionname> [--columnFamily=<columnfamily>] ]", options, true);
+      return false;
+    }
+
+    CommandLineParser parser = new PosixParser();
+    CommandLine cmd = parser.parse(options, args);
+
+    if (cmd.hasOption(fileOption)) {
+      path = new Path(cmd.getOptionValue(fileOption));
+    } else {
+      regionName = cmd.getOptionValue(regionOption);
+      if(StringUtils.isEmpty(regionName)) {
+        err.println("Region name is not specified.");
+        formatter.printHelp("sft [--file=</path/to/tracker/file> | --table=<namespace:tablename|tablename> --region=<regionname> [--columnFamily=<columnfamily>] ]", options, true);
+        System.exit(-1);
+      }
+      columnFamily = cmd.getOptionValue(columnFamilyOption);
+      if(StringUtils.isEmpty(columnFamily)) {
+        err.println("Column family is not specified.");
+        formatter.printHelp("sft [--file=</path/to/tracker/file> | --table=<namespace:tablename|tablename> --region=<regionname> [--columnFamily=<columnfamily>] ]", options, true);
+        System.exit(-1);
+      }
+      String tableNameWtihNS = cmd.getOptionValue(tableNameOption);
+      if(StringUtils.isEmpty(tableNameWtihNS)) {
+        err.println("Table name is not specified.");
+        formatter.printHelp("sft [--file=</path/to/tracker/file> | --table=<namespace:tablename|tablename> --region=<regionname> [--columnFamily=<columnfamily>] ]", options, true);
+        System.exit(-1);
+      }
+      TableName tn = TableName.valueOf(tableNameWtihNS);
+      namespace = tn.getNamespaceAsString();
+      tableName = tn.getNameAsString();
+    }
+    return true;
+  }
+
+  public int run(String[] args) {
+    if(getConf() == null) {
+      throw new RuntimeException("A Configuration instance must be provided.");
+    }
+    try {
+      CommonFSUtils.setFsDefault(getConf(), CommonFSUtils.getRootDir(getConf()));
+      if (!parseOptions(args)) {
+        return 1;
+      }
+    } catch (IOException ex) {
+      LOG.error("Error parsing command-line options", ex);
+      return 1;
+    } catch (ParseException ex) {
+      LOG.error("Error parsing command-line options", ex);
+      return 1;
+    }
+    FileSystem fs = null;
+    if(path != null) {
+      try {
+        fs = path.getFileSystem(getConf());
+        if(fs.isDirectory(path)) {
+          err.println("ERROR, wrong path given: " + path);
+          return -2;
```

Comment:
nit: follow posix standards for return errors (>0)?

Reply:
done